instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -54,9 +57,7 @@ func NewServerHandler(opts ...Option) stats.Handler {
 
 // TagConn can attach some information to the given context.
 func (h *serverHandler) TagConn(ctx context.Context, info *stats.ConnTagInfo) context.Context {
-	span := trace.SpanFromContext(ctx)
-	attrs := peerAttr(peerFromCtx(ctx))
-	span.SetAttributes(attrs...)
+	h.peerAttr = peerAttr(info.RemoteAddr.String())
```

Comment:
How will this work if the Handler has two connections at once?

Reply:
Good question. I will carefully test it tomorrow.

Reply:
I will also double-check that the client handles concurrent calls without races.

Reply:
I also confirm that there can be a race on the client:

```
==================
WARNING: DATA RACE
Write at 0x00c0000a5048 by goroutine 12:
  go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.(*clientHandler).TagConn()
      /home/rpajak/repos/opentelemetry-go-contrib/instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go:137 +0x84

Previous read at 0x00c0000a5048 by goroutine 18:
  go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.(*clientHandler).TagRPC()
      /home/rpajak/repos/opentelemetry-go-contrib/instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go:119 +0x1e7
```

Reply:
I also found out that `TagConn` may be also called after `TagRPC` (or even not at all according to my current testing).

I think that currently it may be impossible to implement.

Reply:
I created https://github.com/grpc/grpc-go/issues/6897